### PR TITLE
[Core] Additional changes for the user @ multiple sites

### PR DESCRIPTION
**Pull request by MounaSafiHarab**
sequel to: https://github.com/aces/Loris/pull/2468 
Changes mainly include grouping user sites in one row in the user_accounts, having the user Site Affiliations count displayed while hovering over shows the sites, and a few bug fixes related to user multisite changes.

### CODE REVIEW

**Comment by MounaSafiHarab** [comment]
@ridz1208 
I assigned to you as you know exactly where these changes are (since you reported some of them).
All in one pull request as changes to each file are minor.

**Comment by ridz1208** [comment]
@MounaSafiHarab line 519 of `modules/issue_tracker/ajax/EditIssue.php` should be addressed

**Comment by MounaSafiHarab** [comment]
@ridz1208 

If you look at my comment in the conversation of this pull request, I have addressed issues 1-5. I have 3 more issues to go (issue tracker is one of them, but I know @alisterdev is making changes to this as well;  you can see the referenced redmine task for this). I might need to issue separate pull requests for the remaining ones, so for now, I hope those changes here can be tested.

**Comment by ridz1208** [comment]
@MounaSafiHarab will do thx

**Comment by ridz1208** [comment]
Below are a few suggestions that would improve UI in my opinion. None are mandatory !!

Suggested changes:
 - no time delay for tooltip to pop open (if possible)
 - tooltip sites to be each on a separate row instead of concatenated
 - "all user site count" I feel is too technical for users maybe "Site associations" ??

**Comment by MounaSafiHarab** [comment]
@ridz1208 

good comments; my answers:
for 1) and 2); this is the functionality we use for tooltip throughout (like for example Configuration), so it was the easiest to do!
3) I will look into displaying an array in the "title" of a tooltip in .tpl (I quickly looked, it is more complicated)
4) I agree. I changed this
5) I agree too, I added space to help a little with the readability (but here again, check certifications in the examiner module; we group_concat this way, without spaces)

@driusan, do you have any input on this? which is necessary you think?

**Comment by MounaSafiHarab** [comment]
@taracampbell  said she will test mainly changes in training/ and dashboard/ as well!

File: modules/user_accounts/php/NDB_Form_user_accounts.class.inc [hunk 1]
```diff
@@ -58,9 +58,11 @@ class NDB_Form_User_Accounts extends NDB_Form
                 if ($this->isCreatingNewUser()) {
                     return true;
                 }
-                return in_array(
-                    $editor->getData('CenterIDs'),
-                    $user->getData('CenterIDs')
+                return array_intersect(
```
Comment (MounaSafiHarab) [inline modules/user_accounts/php/NDB_Form_user_accounts.class.inc]: @driusan 

This part applies when the $editor is not the superuser, and has 'user_accounts' permission but NOT 'user_accounts_multisite'; I am assuming the $editor will be at least part of all the sites of the $user they are editing to return true.

**Comment by taracampbell** [comment]
@MounaSafiHarab - I've looked over the Training changes. I can think of a few issues:

In `markQuiz.php`:
- users can be examiners at multiple sites. This is marked by multiple entries in the examiner table. Right now, when a user passes a quiz, they get certified at one site (`pselectOne`), but we should probably certify them at all of their sites? Not sure about that.

In `NDB_Form_training`:
- The function `getCertificationStatus` checks the certification status for each training module for a user. If there are multiple entries in the examiner table for the user (indicating they are an examiner at multiple sites), this could grab the wrong status if there are discrepancies in the user's certification at different sites

**Comment by MounaSafiHarab** [comment]
@taracampbell 

Thanks! I actually found a problem with my changes whereby I was not "imploding" when getting the centerIDs and using the function `find_in_set`. So this should be fixed now.
 
Regarding your comment, while it is all valid, this has been for a while now that examiners are at multiple sites and their certification is at one site. I still believe the problem is decoupled from the users @ multiple sites. 

@ridz1208 has a few tickets for 17.2 to make additions/changes to the examiner functionality, so this could be one of them, but I will let him comment in more details on this.

**Comment by ridz1208** [comment]
@MounaSafiHarab @taracampbell I just got the chance to review the code and I think we should discuss this. This goes along with the permissions issue where the changes required are related to the user multisite feature but will need to be implemented in a second iteration.

**Comment by MounaSafiHarab** [comment]
Discussed at the PR meeting; @ridz1208 already implemented a fix for the examiner/certification/users. Plan is to have it (in particular, his additional script) tested with a bigger database

**Comment by ridz1208** [comment]
Pending PR for examiner fix (will base it off of this branch)

File: modules/dashboard/php/NDB_Form_dashboard.class.inc [hunk 2]
```diff
@@ -36,7 +36,7 @@ class NDB_Form_Dashboard extends NDB_Form
         $DB     = Database::singleton();
         $user   = User::singleton();
         $config = NDB_Config::singleton();
-        $site   = $user->getSiteName();
+        $site   = explode(';', $user->getSiteName());
```
Comment (driusan) [inline modules/dashboard/php/NDB_Form_dashboard.class.inc]: This doesn't look right. Why is getSiteName returning a semicolon delimited string?

File: smarty/templates/main.tpl [hunk 3]
```diff
@@ -159,11 +159,12 @@
                                 <img width=17 src="{$baseurl}/images/help.gif">
                             </a>
                         </li>
-                        <li>
-                            <p class="navbar-text">
-                                &nbsp;&nbsp;  Sites: {$user.Sites} &nbsp;
-                            </p>
+                        <li class="nav">
+                            <a href="#" data-toggle="tooltip" title="{$user.SitesTooltip}" style="padding-right:25px;">
```
Comment (driusan) [inline smarty/templates/main.tpl]: This shouldn't have inline CSS, it should be part of a CSS class.

File: modules/dashboard/php/NDB_Form_dashboard.class.inc (hunk 2 above)
Comment (MounaSafiHarab) [inline modules/dashboard/php/NDB_Form_dashboard.class.inc]: In the php/libraries/User.class.inc:
I changed the getSiteName() to return array, but it should be left as string as the basic query in the same file is group concatenating the sites (only getCenterID should be changed to return array). So I corrected this.

File: smarty/templates/main.tpl (hunk 3 above)
Comment (MounaSafiHarab) [inline smarty/templates/main.tpl]: @driusan 

I got this from line 169 (a few lines lower). Is it problematic there as well?

**Comment by alisterdev** [comment]
A little contribution from me https://github.com/MounaSafiHarab/Loris/pull/3

**Comment by driusan** [comment]
@MounaSafiHarab This looks fine, but can't be merged because of conflicts.

**Comment by MounaSafiHarab** [comment]
@driusan 

commit message updated. You can now merge. Thanks!

**Pull request merged**